filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsFileSystemProvider.scala

```diff
@@ -201,15 +204,37 @@ class GcsFileSystemProvider private (storageClient: Try[Storage], executionConte
   override def isHidden(path: Path): Boolean = throw new NotImplementedError()
 
   private def list(gcsDir: NioGcsPath) = {
-    val listRequest = client.objects().list(gcsDir.bucket)
+    val maxResults = ConfigFactory.load().getIntOr("google.list-max-results", 1000).toLong
+    val listRequest = client.objects().list(gcsDir.bucket).setMaxResults(maxResults)
     listRequest.setPrefix(gcsDir.objectName)
 
-    val paths = for {
-      listedFile <- withRetry(listRequest.execute()).getItems.asScala
-    } yield NioGcsPath(s"$getScheme${listedFile.getBucket}${GcsFileSystem.Separator}${listedFile.getName}")(gcsDir.getFileSystem.asInstanceOf[GcsFileSystem])
+    def objectToPath(storageObject: StorageObject): Path = {
+      NioGcsPath(s"$getScheme${storageObject.getBucket}${GcsFileSystem.Separator}${storageObject.getName}")(gcsDir.getFileSystem.asInstanceOf[GcsFileSystem])
+    }
+
+    type PageToken = String
```

Comment:
I've really soured on type aliases except for a few key categories. If you want the name is prefer the full belt and suspenders treatment, otherwise just use the string